src/config.rs

```diff
@@ -30,6 +31,7 @@ impl Config {
             editor: None,
             editor_args: None,
             temporary_project_dir,
+            git_repo_depth: None,
```

Comment:
Are you sure you don't want the default to be 1 here?

Reply:
```suggestion
            git_repo_depth: Some(Depth::Level(1)),
```

Reply:
Since git_repo_depth is not needed (as an option), i prefer a None so the first generated config file will be clean